RBAC_Example.groovy

```diff
@@ -0,0 +1,58 @@
+import jenkins.model.Jenkins;
+import nectar.plugins.rbac.strategy.*;
+import hudson.security.*;
+import nectar.plugins.rbac.groups.*;
+import nectar.plugins.rbac.roles.*;
+
+//Obtain security configuration
+RoleMatrixAuthorizationStrategyImpl strategy = RoleMatrixAuthorizationStrategyImpl.getInstance()
+RoleMatrixAuthorizationConfig config = RoleMatrixAuthorizationPlugin.getConfig()
+
+println 'Groups'
+config.getGroups().each{ g ->
+    println '\t' + g.name
+    println '\t\t Group Roles'
+    g.getAllRoles().each{rg -> println '\t\t\t' + rg}
+    println '\t\t Group Memberships'
+    g.getGroupMembership().each{mg -> println '\t\t\t' + mg}
+    println '\t\t Group Members'
+    g.getMembers().each{mg -> println '\t\t\t' + mg}
+    }
+
+println '*Roles*'
+config.getRoles().each{r ->
+    println '\t' + r
+    println '\t\t Role Permissions'
+    Role rc = new Role(r)
+    rc.getPermissionProxies().each{p -> println '\t\t' + p.id + " - " + p.name}
+    }
+
+println '*Permissions*'
+Permission.getAll().each{p -> println '\t' + p.id + " - " + p.name}
+
+println 'create a new Role'
+String roleName = "NewRole"
+strategy.addRole(roleName)
+
+println 'add all permission to NewRole'
+Role rc = new Role(roleName)
+for (Permission p: Permission.getAll()) {
+    if(p.getEnabled() && p.owner == null){
+        rc.doGrantPermissions(p.id)
+    }
+}
+
+println 'remove permission from role'
+rc.doRevokePermissions("hudson.model.Hudson.Read")
+
+println 'create a new group'
```

Comment:
To understand the node structure of Jenkins, here it should be commented: that you are creating the group a **Root** level